PerformanceCalculator/OsuPathUtils.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.IO;
+using Microsoft.Win32;
+
+namespace PerformanceCalculator
+{
+    public class OsuPathUtils
+    {
+        public static string GetOsuPath()
+        {
+            using RegistryKey key = Registry.ClassesRoot.OpenSubKey("osu\\shell\\open\\command");
+            object o = key?.GetValue("");
+
+            if (o == null || !(o is string pathString)) return "";
+
+            pathString = pathString.Split(' ')[0].Trim('"');
+            var pathSlices = pathString.Split(Path.DirectorySeparatorChar);
+            Array.Resize(ref pathSlices, pathSlices.Length - 1);
+            var finalOsuPath = string.Join(Path.DirectorySeparatorChar, pathSlices);
+            return finalOsuPath;
```

Comment:
```suggestion
    try{
            using RegistryKey key = Registry.ClassesRoot.OpenSubKey("osu\\shell\\open\\command");
            object o = key?.GetValue("");

            if (o == null || !(o is string pathString)) return "";

            pathString = pathString.Split(' ')[0].Trim('"');
            var pathSlices = pathString.Split(Path.DirectorySeparatorChar);
            Array.Resize(ref pathSlices, pathSlices.Length - 1);
            var finalOsuPath = string.Join(Path.DirectorySeparatorChar, pathSlices);
            return finalOsuPath;
        }
        catch (Exception ex) {
            return "";
        }
```

`using RegistryKey key = Registry.ClassesRoot.OpenSubKey("osu\\shell\\open\\command");`

this line might cause an exception in case osu\\shell\\open\\command doesn't exist, I haven't tested if it returns null or causes an exception. It's good to have a try-catch here.